### PR TITLE
[MM-15274] Migrate "Team.Get" to Sync by default 

### PR DESCRIPTION
**Pull request by tengis617**
#### Summary

Migrate "Team.Get" to Sync by default 
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/10692

### CODE REVIEW

File: app/command_loadtest.go [hunk 1]
```diff
@@ -215,16 +212,14 @@ func (me *LoadTestProvider) UsersCommand(a *App, args *model.CommandArgs, messag
 		cmd = strings.TrimSpace(strings.TrimPrefix(cmd, "fuzz"))
 	}
 
-	usersr, err := parseRange(cmd, "")
-	if !err {
+	usersr, ok := parseRange(cmd, "")
```
Comment (jespino) [inline app/command_loadtest.go]: nice :)

File: app/team.go [hunk 2]
```diff
@@ -543,11 +556,11 @@ func (a *App) JoinUserToTeam(team *model.Team, user *model.User, userRequestorId
...
 func (a *App) GetTeam(teamId string) (*model.Team, *model.AppError) {
-	result := <-a.Srv.Store.Team().Get(teamId)
-	if result.Err != nil {
-		return nil, result.Err
+	team, err := a.Srv.Store.Team().Get(teamId)
+	if err != nil {
+		return nil, err
 	}
-	return result.Data.(*model.Team), nil
+	return team, nil
```
Comment (jespino) [inline app/team.go]: nit: Probably here you can just replace it with `return a.Srv.Store.Team().Get(teamId)`

File: store/storetest/scheme_store.go [hunk 3]
```diff
@@ -392,9 +392,8 @@ func testSchemeStoreDelete(t *testing.T, ss store.Store) {
 	sres4 := <-ss.Scheme().Delete(d4.Id)
 	assert.Nil(t, sres4.Err)
 
-	tres5 := <-ss.Team().Get(t4.Id)
-	assert.Nil(t, tres5.Err)
-	t5 := tres5.Data.(*model.Team)
+	t5, err := ss.Team().Get(t4.Id)
+	assert.Nil(t, err)
```
Comment (jespino) [inline store/storetest/scheme_store.go]: nit: we can replace it with a `require.Nil` if not we can end up dereferencing a nil pointer.

File: store/storetest/team_store.go [hunk 4]
```diff
@@ -115,7 +115,10 @@ func testTeamStoreUpdateDisplayName(t *testing.T, ss store.Store) {
 		t.Fatal(err)
 	}
 
-	ro1 := (<-ss.Team().Get(o1.Id)).Data.(*model.Team)
+	ro1, err := ss.Team().Get(o1.Id)
+	if err != nil {
```
Comment (jespino) [inline store/storetest/team_store.go]: nit: We can replace this patter for `require.Nil(t, err)` which is more concise and expressive.

File: store/storetest/team_store.go [hunk 5]
```diff
@@ -129,15 +132,16 @@ func testTeamStoreGet(t *testing.T, ss store.Store) {
...
 	store.Must(ss.Team().Save(&o1))
 
-	if r1 := <-ss.Team().Get(o1.Id); r1.Err != nil {
-		t.Fatal(r1.Err)
-	} else {
-		if r1.Data.(*model.Team).ToJson() != o1.ToJson() {
-			t.Fatal("invalid returned team")
-		}
+	r1, err := ss.Team().Get(o1.Id)
+	if err != nil {
```
Comment (jespino) [inline store/storetest/team_store.go]: Same here

File: store/storetest/team_store.go [hunk 6]
```diff
@@ -129,15 +132,16 @@ func testTeamStoreGet(t *testing.T, ss store.Store) {
...
+		t.Fatal(err)
+	}
+
+	if r1.ToJson() != o1.ToJson() {
```
Comment (jespino) [inline store/storetest/team_store.go]: probably we can replace it with an `assert.Equal(t, r1.ToJson(), o1.ToJson())`

File: store/storetest/team_store.go [hunk 7]
```diff
@@ -129,15 +132,16 @@ func testTeamStoreGet(t *testing.T, ss store.Store) {
...
+		t.Fatal("invalid returned team")
 	}
 
-	if err := (<-ss.Team().Get("")).Err; err == nil {
+	if _, err := ss.Team().Get(""); err == nil {
```
Comment (jespino) [inline store/storetest/team_store.go]: We can use the `require.Nil` here too (removing the if)

File: store/storetest/team_store.go [hunk 8]
```diff
@@ -1297,7 +1301,11 @@ func testUpdateLastTeamIconUpdate(t *testing.T, ss store.Store) {
 		t.Fatal(err)
 	}
 
-	ro1 := (<-ss.Team().Get(o1.Id)).Data.(*model.Team)
+	ro1, err := ss.Team().Get(o1.Id)
+	if err != nil {
```
Comment (jespino) [inline store/storetest/team_store.go]: `require.Nil`

File: store/storetest/team_store.go [hunk 9]
```diff
@@ -1472,8 +1480,8 @@ func testResetAllTeamSchemes(t *testing.T, ss store.Store) {
 	res := <-ss.Team().ResetAllTeamSchemes()
 	assert.Nil(t, res.Err)
 
-	t1 = (<-ss.Team().Get(t1.Id)).Data.(*model.Team)
-	t2 = (<-ss.Team().Get(t2.Id)).Data.(*model.Team)
+	t1, _ = ss.Team().Get(t1.Id)
```
Comment (jespino) [inline store/storetest/team_store.go]: add a `require.Nil(t, err)` for each call here to be sure we don't end up deferencing a nil pointer 2 lines later.

File: store/storetest/scheme_store.go (hunk 3 above)
Comment (jespino) [reply store/storetest/scheme_store.go]: this change still needed

**Comment by jespino** [comment]
@tengis617 merge from master needed :)

File: store/storetest/scheme_store.go (hunk 3 above)
Comment (tengis617) [reply store/storetest/scheme_store.go]: @jespino sorry, forgot to push. Now it should be fine.

**Pull request merged**